aten/src/ATen/native/Gesv.cpp

```diff
@@ -72,6 +78,100 @@ static void applyGesv(Tensor& b, Tensor& A, std::vector<int64_t> infos) {
   }
 }
 
+std::tuple<Tensor&,Tensor&> _gesv_single_out_cpu(
+    Tensor& sol, Tensor& lu,
+    const Tensor& self, const Tensor& A) {
+#ifndef USE_LAPACK
+  AT_ERROR("gesv: LAPACK library not found in compilation");
+#endif
+  int64_t bx = self.size(0);
+  int64_t by = (self.dim() == 1) ? 1 : self.size(1);
+  int64_t ax = A.size(0);
+  int64_t ay = A.size(1);
+  int info = 0;
+
+  bool use_temp_sol = false;
+  bool use_temp_lu = false;
+  auto temp_sol = self.type().tensor();
```

Comment:
Can you not allocate temporary tensors if not necessary?

Reply:
Switched to at::optional

Reply:
Since `at::optional` didn't work with MSVC, switched to nullable Tensors and using `t.defined()` to do the appropriate allocations